x-pack/plugins/infra/public/containers/logs/log_position/use_log_position.ts

```diff
@@ -71,65 +54,26 @@ type UpdateDateRangeFn = (
 const DESIRED_BUFFER_PAGES = 2;
 const RELATIVE_END_UPDATE_DELAY = 1000;
 
-export const useLogPositionState: () => LogPositionStateParams & LogPositionCallbacks = () => {
-  const { initialStateFromUrl, startSyncingWithUrl } = useLogPositionUrlStateSync();
-  const { initialStateFromTimefilter, startSyncingWithTimefilter } =
-    useLogPositionTimefilterStateSync();
-
-  const [logPositionStateContainer] = useState(() =>
-    withReduxDevTools(
-      createLogPositionStateContainer({
-        initialStateFromUrl,
-        initialStateFromTimefilter,
-      }),
-      {
-        name: 'logPosition',
-      }
-    )
-  );
-
-  useEffect(() => {
-    return startSyncingWithUrl(
-      wrapStateContainer<LogPositionState, LogPositionUrlState>({
-        wrapGet: getUrlState,
-        wrapSet: updateStateFromUrlState,
-      })(logPositionStateContainer)
-    );
-  }, [logPositionStateContainer, startSyncingWithUrl]);
-
-  useEffect(() => {
-    return startSyncingWithTimefilter(
-      wrapStateContainer<LogPositionState, TimefilterState>({
-        wrapGet: getTimefilterState,
-        wrapSet: updateStateFromTimefilterState,
-      })(logPositionStateContainer)
-    );
-  }, [logPositionStateContainer, startSyncingWithTimefilter, startSyncingWithUrl]);
-
-  const { latestValue: latestLogPositionState } = useObservableState(
-    logPositionStateContainer.state$,
-    () => logPositionStateContainer.get()
-  );
-
-  const dateRange = useMemo(
-    () => getLegacyDateRange(latestLogPositionState),
-    [latestLogPositionState]
-  );
-
-  const { targetPosition, visiblePositions } = latestLogPositionState;
+export const useLogPositionState = ({
+  logStreamPageState,
+  logStreamPageSend,
+}: {
+  logStreamPageState: InitializedLogStreamPageState;
+  logStreamPageSend: LogStreamPageSend;
+}): LogPositionStateParams & LogPositionCallbacks => {
+  const dateRange = useMemo(() => getLegacyDateRange(logStreamPageState), [logStreamPageState]);
 
-  const isStreaming = useMemo(
-    () => !latestLogPositionState.refreshInterval.pause,
-    [latestLogPositionState]
-  );
+  const { refreshInterval, timeRange, targetPosition, visiblePositions, latestPosition } =
+    logStreamPageState.context;
+  const isStreaming = useMemo(() => !refreshInterval.pause, [refreshInterval]);
```

Comment:
**nit:** I see this was already memorized and it's not a new change, but since it memorizes just a boolean value, here `useMemo` seems more like a de-optimization. I didn't measure it, it's a theoretic assumption, feel free to ignore this if there are good reasons to memorize this.

Reply:
Good point. There's no reason here, it's just brought over from the old code. Again, probably a micro-improvement but since it's an easy win I'll change it over 👍